index.bs

```diff
@@ -1129,13 +1138,68 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
+
+                          :   is set to {{ResidentKeyRequirement/preferred}}
+                          ::  If the |authenticator| is not capable of storing a [=client-side-resident public key credential
+                              source=], and there are other known authenticators attached that are capable of storing a
+                              [=client-side-resident public key credential source=], [=iteration/continue=].
+
+                          :   is set to {{ResidentKeyRequirement/discouraged}}
+                          ::  If the |authenticator| is not capable of creating a [=server-side-resident public key credential
+                              source=], and there are other known authenticators attached that are capable of creating a
+                              [=server-side-resident public key credential source=], [=iteration/continue=].
+
+                          :   is [=not present=]
+                          ::  if <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{requireResidentKey}}</code>
+                              is set to [TRUE] and the |authenticator| is not capable of storing a [=client-side-resident public
+                              key credential source=], [=iteration/continue=].
+                      </dl>
+
                   1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{AuthenticatorSelectionCriteria/userVerification}}</code> is
                     set to {{UserVerificationRequirement/required}} and the |authenticator| is not capable of performing [=user
                     verification=], [=iteration/continue=].
 
+            1. Let |requireResidentKey| be the <dfn>effective resident key requirement for credential creation</dfn>, a Boolean value,
+                as follows. 
+
+                If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{residentKey}}</code>
+
+                    <dl class="switch">
+
+                        :   is set to {{ResidentKeyRequirement/discouraged}}
+                        ::  Let |requireResidentKey| be [FALSE].
+
+                        :   is set to {{ResidentKeyRequirement/preferred}}
+                        ::  If the |authenticator|
+
+                            <dl class="switch">
+                                :   is capable of [=client-side-resident public key credential source=]
+                                ::  Let |requireResidentKey| be [TRUE].
+
+                                :   is not capable of [=client-side-resident public key credential source=], or if the client cannot determine authenticator capability,
+                                ::  Let |requireResidentKey| be [FALSE].
+                            </dl>
+
+                        :   is set to {{ResidentKeyRequirement/required}}
+                        ::  Let |requireResidentKey| be [TRUE].
+
+                        :   is [=not present=]
```

Comment:
(Ditto about a default value excluding this case.)

Reply:
Understood - removed default value for requireResidentKey